### ISSUE TITLE
`unicorn/prefer-regexp-test` false positive with string regexp

### ISSUE THREAD

**Issue by bmish** [sindresorhus/eslint-plugin-unicorn#1150]
The rule has a false positive / incorrect autofix when using a string regexp.

eslint-plugin-unicorn v19 / [unicorn/prefer-regexp-test](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v29.0.0/docs/rules/prefer-regexp-test.md)

Before autofix:

```js
const endsWithPunctuationRegex = '[.!?]\\s*$';
const myString = 'hello world';
if (myString.match(endsWithPunctuationRegex)) {
  console.log('found match');
}
```

After autofix:

```js
const endsWithPunctuationRegex = '[.!?]\\s*$';
const myString = 'hello world';
if (endsWithPunctuationRegex.test(myString)) {
  console.log('found match');
}
```

Result:

```
Uncaught TypeError: endsWithPunctuationRegex.test is not a function
```

**Comment by fisker** [comment]
Yah, it's bug, when I implement this rule, I didn't know `String#match()` accept `String`, but later found people actually use this somewhere.
